docs/design/2021-03-09-dynamic-privileges.md

```diff
@@ -0,0 +1,348 @@
+| GRANT SELECT, INSERT, UPDATE, DELETE, CREATE, DROP, RELOAD, SHUTDOWN, PROCESS, FILE, REFERENCES, INDEX, ALTER, SHOW DATABASES, SUPER, CREATE TEMPORARY TABLES, LOCK TABLES, EXECUTE, REPLICATION SLAVE, REPLICATION CLIENT, CREATE VIEW, SHOW VIEW, CREATE ROUTINE, ALTER ROUTINE, CREATE USER, EVENT, TRIGGER, CREATE TABLESPACE, CREATE ROLE, DROP ROLE ON *.* TO `u1`@`%`                                                                                                                                                                                                       |
+| GRANT APPLICATION_PASSWORD_ADMIN,AUDIT_ADMIN,BINLOG_ADMIN,BINLOG_ENCRYPTION_ADMIN,CLONE_ADMIN,CONNECTION_ADMIN,ENCRYPTION_KEY_ADMIN,FLUSH_OPTIMIZER_COSTS,FLUSH_STATUS,FLUSH_TABLES,FLUSH_USER_RESOURCES,GROUP_REPLICATION_ADMIN,INNODB_REDO_LOG_ARCHIVE,INNODB_REDO_LOG_ENABLE,PERSIST_RO_VARIABLES_ADMIN,REPLICATION_APPLIER,REPLICATION_SLAVE_ADMIN,RESOURCE_GROUP_ADMIN,RESOURCE_GROUP_USER,ROLE_ADMIN,SERVICE_CONNECTION_ADMIN,SESSION_VARIABLES_ADMIN,SET_USER_ID,SHOW_ROUTINE,SYSTEM_USER,SYSTEM_VARIABLES_ADMIN,TABLE_ENCRYPTION_ADMIN,XA_RECOVER_ADMIN ON *.* TO `u1`@`%` |
+| GRANT BACKUP_ADMIN ON *.* TO `u1`@`%` WITH GRANT OPTION                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |
++------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+3 rows in set (0.00 sec)
+```
+
+Currently TiDB does not expand `GRANT ALL` when the value is read back from `SHOW GRANTS`. It is possible to maintain this current behavior difference.
+
+#### Alter User
+
+No change
+
+### Initial Set of Dynamic Privileges
+
+#### Borrowed from MySQL
+
+| Privilege Name | Description | Notes |
+| --------------- | --------------- | --------------- |
+| `BACKUP_ADMIN` | Enables BR backups and restores, as well as lightning restores. | Currently this required `SUPER`. It will now require `BACKUP_ADMIN` or `SUPER`. |
```

Comment:
If SEM is enabled, can `SUPER` backup and restore? There seems no document defining this behavior.

Reply:
Unless it is listed under https://github.com/pingcap/tidb/pull/23223/files#diff-44d2981b55b0771725cf2ef4bd2b3e0eac71fe2124bc05dcb704e3fbf0cfbdf0R46 in SEM, it continues to work the same with/without SEM enabled. So the current description here is that it will require `BACKUP_ADMIN` or `SUPER`.

What is not explicitly clarified (but I will add to the SEM document now) is that dynamic privileges of name `RESTRICTED_*` will not accept `SUPER` as an alternative.

Reply:
@SunRunAway see https://github.com/pingcap/tidb/pull/23223/commits/e158e80efcae0f8c87710110d47cfba31b8eacc5 which makes this clearer.

Reply:
Thanks. It Makes sense.